containers/Dockerfile.buildenv

```
@@ -38,8 +38,29 @@
 # If you do not have permission to push to DockerHub please coordinate with
 # someone who does. Since changes to this image effect our testing
 # infrastructure it is important all changes be carefully reviewed.
+#
+# NOTE: If building for Apple Silicon, then add the following arguments to
+# the docker build command in step 1 and in step 3 (building the ci target)
+# is not currently supported, since GitHub actions do not yet support
+# Apple Silicon:
+#
+#  --build-arg "CHARM_ARCH=arm8" --build-arg "TEX_ARCH=aarch64"
+#
 
-FROM ubuntu:20.04 AS dev
+FROM --platform=linux/$BUILDARCH ubuntu:20.04 AS dev
```

Avoid the extra `CHARM_ARCH` and `TEX_ARCH` build args like this:

```suggestion
FROM ubuntu:20.04 AS base

ARG TARGETARCH

FROM base AS base-amd64
ENV CHARM_ARCH=x86_64
ENV TEX_ARCH=x86_64

FROM base AS base-arm64
ENV CHARM_ARCH=arm8
ENV TEX_ARCH=aarch64

FROM base-${TARGETARCH} AS dev
```